### PR TITLE
TETP test admin posting form page

### PR DESCRIPTION
**Pull request by mattikl**

### CODE REVIEW

File: frontend/tet/admin/src/__tests__/utils/components/get-editor-api.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    inputValueIsPresent: async <K extends keyof TetPosting>(key: K): Promise<void> => {
+      const field = await screen.findByTestId(`posting-form-${key}`);
+      if (!expectedPosting) {
+        throw new Error('you forgot to give expected application values for the test');
+      }
+      const value = expectedPosting[key] as string;
+      expect(field).toHaveValue(value);
+    },
+    languageValuesArePresent: async (): Promise<void> => {
+      const select = await screen.findByText(/Tet-jaksolla käytetty kieli/i);
```
Comment (nenonja) [inline frontend/tet/admin/src/__tests__/utils/components/get-editor-api.ts:25]: use translation key?

File: frontend/tet/admin/src/__tests__/utils/components/get-editor-api.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    },
+    languageSelectorHasError: async (): Promise<void> => {
+      const field = await screen.findByRole('button', {
+        name: regexp(translations.editor.posting.contactLanguage),
+      });
+      const parent = field?.parentElement?.parentElement;
+      await within(parent).findByText(requiredText);
+    },
+    selectionGroupHasError: async (labelText: string): Promise<void> => {
+      await screen.findByText(regexp('Työtapa'));
```
Comment (nenonja) [inline frontend/tet/admin/src/__tests__/utils/components/get-editor-api.ts:50]: use translation key?

File: frontend/tet/admin/src/components/editor/__tests__/Editor.test.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      expectWorkingMethodsFromLinkedEvents();
+      expectAttributesFromLinkedEvents();
+
+      renderComponent(<Editor />);
+      const editorApi = getEditorApi(posting);
+      await editorApi.actions.clickSendButton();
+
+      await waitForBackendRequestsToComplete();
+
+      await screen.findByRole('heading', {
```
Comment (nenonja) [inline frontend/tet/admin/src/components/editor/__tests__/Editor.test.tsx:89]: move this to editor api expectations and use translation key?

**Review by nenonja — COMMENTED** [review]
Looks good! Now just have to make the tough work to make the all tests to pass...
There will be a lot less warnings and other spams after the testing-library update pr gets in.
Also with these type of errors:
```
 // eslint-disable-next-line testing-library/prefer-find-by
      21 |     await waitFor(() => {
    > 22 |       expect(nock.isDone()).toBeTruthy();
         |                             ^
      23 |     });
```
It probably means that you have to add to some test(s) the `waitForBackendRequestsToComplete()` somewhere (preferably before checking the expected results)

**Review by pekkvil — APPROVED** [review]

**Pull request merged**